hs_core/hydroshare/resource.py

```diff
@@ -768,20 +768,18 @@ def add_resource_files(pk, *files, **kwargs):
     fed_zone_home_path = kwargs.pop('fed_zone_home_path', '')
     # for adding files to existing resources, the default action is copy
     fed_copy_or_move = kwargs.pop('fed_copy_or_move', 'copy')
+    folder = kwargs.pop('folder', None) 
 
-    for f in files:
+    for f in files: 
         if fed_zone_home_path:
             # user has selected files from a federated iRODS zone, so files uploaded from local disk
-            # need to be stored to the federated iRODS zone rather than HydroShare zone. A typical
-            # case is when user selected one file from federated zone, and select one file from
-            # local disk, in which case both files will be added to the federated user zone
-            ret.append(utils.add_file_to_resource(resource, f,
-                                                  fed_res_file_name_or_path=fed_zone_home_path))
+            # need to be stored to the federated iRODS zone rather than HydroShare zone as well
+            ret.append(utils.add_file_to_resource(resource, f, folder=folder, fed_res_file_name_or_path=fed_zone_home_path))
         elif resource.resource_federation_path:
             # file needs to be added to a resource in a federated zone
             ret.append(utils.add_file_to_resource(resource, f, fed_res_file_name_or_path=resource.resource_federation_path))
```

Comment:
My guess is that further modifications in my pipeline are going to make this much easier. I spent a week making sure that federated paths work out properly. So, I think it would be better to wait for the next PR before spending much effort on testing.